### PR TITLE
5D Batchnorm

### PR DESCRIPTION
**Pull request by mys007**
Modified the code to allow volumetric batchnorm, as cudnn has native support for it.

### CODE REVIEW

**Comment by mys007** [comment]
Oh, there is already https://github.com/soumith/cudnn.torch/pull/94, sorry. My PR extends SpatialBatchNormalization, while the other PR creates a new class VolumetricBatchNormalization. I'll leave the decision up to the illuminati.

**Comment by misko** [comment]
I duplicated most of the code from spatial batchnorm to make the volumetric batchnorm class because I wasn't sure if having it in the same class would cause confusion for someone trying to use non-batched mode 3D inputs (4D tensor). Even though the user should be aware of this before using the function... ? 

Also almost all of the code for spatial batch norm in the NN implementation seems to extend to volumetric with minor changes, pretty much just changing one line in the implementation. https://github.com/torch/nn/pull/642

**Comment by szagoruyko** [comment]
I think it should be VolumetricBatchNormalization subclassed from cudnn.SpatialBatchNormalization with Martin's `SpatialBatchNormalization:createIODescriptors` function

**Comment by soumith** [comment]
@colesbury is implementing the Volumetric in nn. Will merge this PR (or @misko's PR ) within the week, after that one is merged into nn.

**Comment by soumith** [comment]
closed via https://github.com/soumith/cudnn.torch/pull/124
